convert2rhel/backup.py

```diff
@@ -459,3 +459,4 @@ def remove_epoch_from_yum_nevra_notation(package_nevra):
 
 changed_pkgs_control = ChangedRPMPackagesController()  # pylint: disable=C0103
 backup_control = BackupController()
+package_files_changes = []  # control of changed package files
```

Comment:
I had problems with storing this list in the action framework. So since it's related to backup I put it there.

Reply:
Once we're storing state, it feels like we should be using a `RestorableChange` (Writing a new subclass of that) for this instead of adding a new global.  It might be easiest to port `backup.RestorableFile`'s functionality into a `RestorableChange` and then for each file that we determine we need to backup and restore, create a RestorableFile and push that onto the backup_control instance.

Reply:
@abadger I totally agree we should use the `RestorableChange`. But I didn't want to port the `RestorableFile` in this PR, since it's in a different task. My idea was to write it using the old way and then port it in the task for porting (which sounds terrible to me when I'm writing it :D). 

What just came to my mind is to port the `RestorableFile` there and leave the old one too. Later we can port the things that are using the old way to the new `RestorableFile` which would be already prepared. So from my point of view the "new" `RestorableFile` can be under this PR since it would be used just for this feature.

Another idea is to have PR just for porting the `RestorableFile` class, but without porting the usage of it - this could be done in some other PR. Maybe this would be the best to have the PRs simple - we can just rebase the dependent ones. This is mostly the same as the previous, but maybe more systematic?

My motivation for this is not to do many changes and have the PR as simple as can be. And also not falling into some cherry-picking etc. between PRs (I remember it from the PRs before last merge freeze which I reviewed) and have to worry about some sync between them.

What do you think? What would be the best way to do this?

Reply:
Your second idea is the cleanest: 

> Another idea is to have PR just for porting the RestorableFile class, but without porting the usage of it - this could be done in some other PR. Maybe this would be the best to have the PRs simple - we can just rebase the dependent ones. This is mostly the same as the previous, but maybe more systematic?

But your first idea is okay as well:

> What just came to my mind is to port the RestorableFile there and leave the old one too. Later we can port the things that are using the old way to the new RestorableFile which would be already prepared. So from my point of view the "new" RestorableFile can be under this PR since it would be used just for this feature.

The problems in the last release stemmed from starting to do the second one but not keeping track of where the changes needed to go, which PR needed to be reviewed first, and not concentrating on getting the base PR merged ASAP.  We had a secondary problem of the dependent PRs all bein fairly large changes in and of themselves.

I think either of your plans for this one will avoid those pitfalls.

Reply:
The second idea is done there: https://github.com/oamg/convert2rhel/pull/935